api/bin/chainflip-broker-api/src/main.rs

```diff
@@ -296,10 +302,10 @@ impl RpcServer for RpcServerImpl {
 
 	async fn register_affiliate(
 		&self,
-		affiliate_id: AccountId32,
-		short_id: Option<AffiliateShortId>,
-	) -> RpcResult<AffiliateShortId> {
-		Ok(self.api.broker_api().register_affiliate(affiliate_id.clone(), short_id).await?)
+		short_id: AffiliateShortId,
+		withdrawal_address: AddressString,
```

Comment:
Hmm... not sure we need to provide an id here. Otherwise the broker needs to query the chain first, to find out what ids are taken.

Previously it was possible re-assign an id, so it kinda made sense, but not any more.

We only expect this to be used for new ids, so we could just generate one, as long as the provided withdrawal address hasn't been registered already. Then we return the generated `AffiliateDetails`.